cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -1240,6 +1240,19 @@ bool CacheAllocator<CacheTrait>::moveChainedItem(ChainedItem& oldItem,
   return true;
 }
 
+template <typename CacheTrait>
+void CacheAllocator<CacheTrait>::unlinkItemForEviction(Item& it) {
+  XDCHECK(it.isMarkedForEviction());
+  XDCHECK(it.getRefCount() == 0);
+  accessContainer_->remove(it);
+  removeFromMMContainer(it);
```

Comment:
didn't we already remove this item from mm-container in findEviction at L1313? (if it were a parent or just a regular item?)

Reply:
Yes, we might have removed the item but `removeFromMMContainer` checks if the item is still in the MMContainer and only then calls mmcontainer.remove(). We need to keep this call here because If the parent of the item changed, the item will still be in the mmcontainer.